### DISCUSSION TITLE
Multi-provider transaction submit endpoint

### DISCUSSION THREAD

**Discussion by mmahut** [blockfrost/blockfrost-backend-ryo#8] [Ideas]
Today, Blockfrost's submit servers are processing a large percentage of network transactions and we have seen occasion where it has been over 50% of all transaction in an epoch, which is not optimal from decentralization point of view.

Clients have right now the possibility to use a different provider to submit their transactions, but from the numbers we have seen this is a minority.

We should implement a multi-provider transaction submit endpoint to the Blockfrost RYO backend, that would support providers such as:

* cardano-submit-api
* Ogmios
* cardano-graphql
* Freeloaders

This new endpoint structure needs to provide an easy integration path for new upcoming ways (and providers) of submitting transactions.

**Comment by profd2004** [comment]
This is a great idea and happy for the conversation and education once we have a pattern, and implementation. From a non-architect point of view, seems like this is a network challenge? No concrete ideas popping into my head at the moment. Love that that conversion is happening!

**Comment by pbwebdev** [comment]
I'm trying to understand the flow of a transaction from Blockfrost API to the providers.

Would the Blockfrost API route requests to different providers based on the parameters we set in our requests?

**Reply by mmahut** [reply]
Ideally, we would need two things to improve the situation:

- A new plugin in [blockfrost-backend-ryo](https://github.com/blockfrost/blockfrost-backend-ryo) that will allow developers to choose the provider
- A new service option on hosted blockfrost.io where the user will be able to choose which provider will their project use (the system will relay this txs to the provider of choice)
  - But also an incentive structure to invite independent SPOs to join the effort and provide their submit nodes

**Comment by mmahut** [comment]
We have a catalyst proposal to implement this feature. https://cardano.ideascale.com/c/idea/107608
